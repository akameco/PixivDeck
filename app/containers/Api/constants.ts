import { API_REQUEST_SUCCESS_TYPE } from './actionTypes'

export const API_REQUEST_SUCCESS: API_REQUEST_SUCCESS_TYPE =
  'Api/API_REQUEST_SUCCESS'
