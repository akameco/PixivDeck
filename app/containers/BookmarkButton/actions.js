// @flow
import type { Action } from './actionTypes'
import {
  ADD_BOOKMARK_REQUEST,
  ADD_BOOKMARK_SUCCESS,
  DELETE_BOOKMARK_REQUEST,
  DELETE_BOOKMARK_SUCCESS,
  ADD_BOOKMARK_FAILER,
  DELETE_BOOKMARK_FAILER,
} from './constants'
import type { Restrict } from './types'

export function addBookmarkRequest(id: number, restrict: Restrict): Action {
  return {
    type: ADD_BOOKMARK_REQUEST,
    id,
    restrict,
  }
}

export function addBookmarkSuccess(id: number, restrict: Restrict): Action {
  return {
    type: ADD_BOOKMARK_SUCCESS,
    id,
    restrict,
  }
}

export function deleteBookmarkRequest(id: number): Action {
  return {
    type: DELETE_BOOKMARK_REQUEST,
    id,
  }
}

export function deleteBookmarkSuccess(id: number): Action {
  return {
    type: DELETE_BOOKMARK_SUCCESS,
    id,
  }
}

export function addBookmarkFailer(id: number, error: string): Action {
  return {
    type: ADD_BOOKMARK_FAILER,
    id,
    error,
  }
}

export function deleteBookmarkFailer(id: number, error: string): Action {
  return {
    type: DELETE_BOOKMARK_FAILER,
    id,
    error,
  }
}
