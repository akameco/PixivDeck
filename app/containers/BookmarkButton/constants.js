// @flow
import type {
  ADD_BOOKMARK_REQUEST_TYPE,
  ADD_BOOKMARK_SUCCESS_TYPE,
  ADD_BOOKMARK_FAILER_TYPE,
  DELETE_BOOKMARK_REQUEST_TYPE,
  DELETE_BOOKMARK_SUCCESS_TYPE,
  DELETE_BOOKMARK_FAILER_TYPE,
} from './actionTypes'

export const ADD_BOOKMARK_REQUEST: ADD_BOOKMARK_REQUEST_TYPE =
  'BookmarkButton/ADD_BOOKMARK_REQUEST'
export const ADD_BOOKMARK_SUCCESS: ADD_BOOKMARK_SUCCESS_TYPE =
  'BookmarkButton/ADD_BOOKMARK_SUCCESS'
export const ADD_BOOKMARK_FAILER: ADD_BOOKMARK_FAILER_TYPE =
  'BookmarkButton/ADD_BOOKMARK_FAILER'
export const DELETE_BOOKMARK_REQUEST: DELETE_BOOKMARK_REQUEST_TYPE =
  'BookmarkButton/ELETE_BOOKMARK_REQUEST'
export const DELETE_BOOKMARK_SUCCESS: DELETE_BOOKMARK_SUCCESS_TYPE =
  'BookmarkButton/DELETE_BOOKMARK_SUCCESS'
export const DELETE_BOOKMARK_FAILER: DELETE_BOOKMARK_FAILER_TYPE =
  'BookmarkButton/DELETE_BOOKMARK_FAILER'
