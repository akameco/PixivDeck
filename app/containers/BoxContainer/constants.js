// @flow
import type { OPEN_PIXIV_TYPE } from './actionTypes'

export const OPEN_PIXIV: OPEN_PIXIV_TYPE = 'BoxContainer/OPEN_PIXIV'
