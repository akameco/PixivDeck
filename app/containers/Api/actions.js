// @flow
import type { Response } from 'services/api'
import type { Action } from './actionTypes'
import { API_REQUEST_SUCCESS } from './constants'

export function apiRequestSuccess(response: Response): Action {
  return {
    type: API_REQUEST_SUCCESS,
    response,
  }
}
