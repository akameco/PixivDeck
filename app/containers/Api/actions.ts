import { Response } from 'services/api'
import { Action } from './actionTypes'
import { API_REQUEST_SUCCESS } from './constants'

export function apiRequestSuccess(response: Response): Action {
  return {
    type: API_REQUEST_SUCCESS,
    response,
  }
}
