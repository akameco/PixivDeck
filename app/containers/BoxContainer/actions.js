// @flow
import type { Action } from './actionTypes'
import { OPEN_PIXIV } from './constants'

export function openPixiv(id: number): Action {
  return {
    type: OPEN_PIXIV,
    id,
  }
}
