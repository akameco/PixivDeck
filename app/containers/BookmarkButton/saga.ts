import { call, put, takeEvery } from 'redux-saga/effects'
import * as api from 'containers/Api/sagas'
import { get } from '../Api/sagas'
import * as columnActions from '../ColumnBookmark/actions'
import * as Actions from './constants'
import * as actions from './actions'
import { Restrict } from './types'

interface Props {
  id: number
  restrict: Restrict
}
export function* bookmark({ id, restrict }: Props) {
  try {
    yield call(api.post, '/v2/illust/bookmark/add', {
      illustId: id,
      restrict,
    })
    yield put(actions.addBookmarkSuccess(id, restrict))
  } catch (error) {
    yield put(actions.addBookmarkFailer(id, error))
  }
}
export function* deleteTask({ id }: Props) {
  try {
    yield call(api.post, '/v1/illust/bookmark/delete', {
      illustId: id,
    })
    yield put(actions.deleteBookmarkSuccess(id))
    yield put(columnActions.removeItem('public', id))
  } catch (error) {
    yield put(actions.deleteBookmarkFailer(id, error))
  }
}

function* success({ id }: Props) {
  try {
    yield call(get, `/v1/illust/detail?illust_id=${id}`, true)
  } catch (error) {}
}

function* root() {
  yield takeEvery(Actions.ADD_BOOKMARK_REQUEST, bookmark)
  yield takeEvery(Actions.DELETE_BOOKMARK_REQUEST, deleteTask)
  yield takeEvery(
    [Actions.ADD_BOOKMARK_SUCCESS, Actions.DELETE_BOOKMARK_SUCCESS],
    success
  )
}

export default root
