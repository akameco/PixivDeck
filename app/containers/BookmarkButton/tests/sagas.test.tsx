import { takeEvery, put } from 'redux-saga/effects'
import * as sagas from '../saga'
import * as constants from '../constants'
import * as actions from '../actions'

test('root', () => {
  const gen = sagas.default()
  const next = gen.next()
  expect(next.value).toStrictEqual(
    takeEvery(constants.ADD_BOOKMARK_REQUEST, sagas.bookmark)
  )
})

test('bookmark', () => {
  const action = {
    id: 1,
    restrict: 'public',
  }
  const gen = sagas.bookmark(action)
  let next = gen.next()
  expect(next.value).toMatchSnapshot()
  next = gen.next()
  expect(next.value).toStrictEqual(put(actions.addBookmarkSuccess(1, 'public')))
})

test('bookmark failed', () => {
  const action = {
    id: 1,
    restrict: 'public',
  }
  const gen = sagas.bookmark(action)
  let next = gen.next()
  expect(next.value).toMatchSnapshot()
  next = gen.throw('error')
  expect(next.value).toStrictEqual(put(actions.addBookmarkFailer(1, 'error')))
})
