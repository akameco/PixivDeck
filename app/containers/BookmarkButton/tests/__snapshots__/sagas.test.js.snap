// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`bookmark 1`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [
      "/v2/illust/bookmark/add",
      Object {
        "illustId": 1,
        "restrict": "public",
      },
    ],
    "context": null,
    "fn": [Function],
  },
  "type": "CALL",
}
`;

exports[`bookmark failed 1`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [
      "/v2/illust/bookmark/add",
      Object {
        "illustId": 1,
        "restrict": "public",
      },
    ],
    "context": null,
    "fn": [Function],
  },
  "type": "CALL",
}
`;
